Converter/igtlioNDArrayConverter.cxx

```diff
@@ -0,0 +1,51 @@
+#include "igtlioNDArrayConverter.h"
+#include "igtlNDArrayMessage.h"
+#include <igtl_util.h>
+#include <vtkDataArray.h>
+
+int igtlioNDArrayConverter::fromIGTL(igtl::MessageBase::Pointer source,
+                                            HeaderData* header,
+                                            ContentData* dest,
+                                            bool checkCRC,
+                                            igtl::MessageBase::MetaDataMap& outMetaInfo)
+{
+    igtl::NDArrayMessage::Pointer msg;
+    msg = igtl::NDArrayMessage::New();
+    msg->Copy(source);
+
+    int c = msg->Unpack(checkCRC);
+    
+
+    cbcbcpa
+    if ((c & igtl::MessageHeader::UNPACK_BODY == 0))
+        {
+        return 0;
+        }
+
+    if (!IGTLtoHeader(dynamic_pointer_cast<igtl::MessageBase>(msg), header, outMetaInfo))
+        return 0;
+
+    vtkSmartPointer<vtkDataArray> NDArray_msg = dest->NDArray_msg;
```

Comment:
This shouldn't be copy. Create a new empty pointer
`vtkSmartPointer<vtkDataArray> NDArray_msg = vtkSmartPointer<vtkDataArray>::New();`

Reply:
More than that. You'll have to switch on the IGTL array scalar type so you create the correct type of array (vtkFloatArray, vtkIntArray, etc...)